addons/mail/models/mail_thread.py

```diff
@@ -1409,6 +1409,9 @@ def _message_parse_extract_payload(self, message, save_original=False):
                     alternative = True
                 if part.get_content_type() == 'multipart/mixed':
                     mixed = True
+                if part.get_content_type() == 'bin/plain' and 'attachment' in part.get('Content-Disposition'):
+                    _logger.warning("Message containing an unexpected Content-Type, assuming 'attachment'")
+                    part.replace_header('Content-Type', 'application/pdf;' + part.get('Content-Type', '').split(';')[1])
```

Comment:
you could have any number of semi-columns. Something like `part.get('ContentType', '').replace('bin/plain', 'application/pdf') seems more appropriate`